lib/policies/oauth2/oauth2.js

```diff
@@ -172,6 +171,9 @@ module.exports = function () {
 
 function setHeaders (req, headerObject, namespace) {
   for (let key in headerObject) {
+    if (key === 'tokenDecrypted') {
```

Comment:
I do not really like to couple this specific logic here. `setHeaders` should just set the headers. Providing the right ones should be a caller responsibility.